mr-hive3/src/main/java/org/apache/iceberg/mr/hive/serde/objectinspector/IcebergDateObjectInspector.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.mr.hive.serde.objectinspector;
+
+import java.time.LocalDate;
+import org.apache.hadoop.hive.common.type.Date;
+import org.apache.hadoop.hive.serde2.io.DateWritableV2;
+import org.apache.hadoop.hive.serde2.objectinspector.primitive.AbstractPrimitiveJavaObjectInspector;
+import org.apache.hadoop.hive.serde2.objectinspector.primitive.DateObjectInspector;
+import org.apache.hadoop.hive.serde2.typeinfo.TypeInfoFactory;
+import org.apache.iceberg.util.DateTimeUtil;
+
+public final class IcebergDateObjectInspector extends AbstractPrimitiveJavaObjectInspector
+                                              implements DateObjectInspector {
+
+  private static final IcebergDateObjectInspector INSTANCE = new IcebergDateObjectInspector();
+
+  public static IcebergDateObjectInspector get() {
+    return INSTANCE;
+  }
+
+  private IcebergDateObjectInspector() {
+    super(TypeInfoFactory.dateTypeInfo);
+  }
+
+  @Override
+  public Date getPrimitiveJavaObject(Object o) {
+    if (o == null) {
+      return null;
+    }
+    LocalDate date = (LocalDate) o;
+    return Date.ofEpochDay((int) date.toEpochDay());
```

Comment:
Why use `toEpochDay` here, but `daysFromDate` in the other method? I think both should use `daysFromDate` since that's what we use elsewhere and trust the most.

Reply:
Good point, thanks, I will change to use the same method